SimplenoteUITests/NoteEditor.swift

```diff
@@ -96,35 +96,35 @@ class NoteEditor {
     }
 
     class func getCheckboxesForTextCount(text: String) -> Int {
-        let matchesCount = NoteEditor.getTextViewsWithExactLabelCount(label: text)
-        print(">>> ^ Found " + String(matchesCount) + " Checkboxe(s) for '" + text + "'")
-        return matchesCount
+        let matches = NoteEditor.getTextViewsWithExactLabelCount(label: text)
+        print(">>> ^ Found " + String(matches) + " Checkbox(es) for '" + text + "'")
+        return matches
     }
 
     class func getTextViewsWithExactValueCount(value: String) -> Int {
         let textViews = getAllTextViews()
-        var matchesCounter = 0
+        var matches = 0
 
-        for index in 0...textViews.count - 1 {
-            let currentValue = textViews.element(boundBy: index).value as! String
+        for textView in textViews {
+            let currentValue = textView.value as! String
             let currentValueStripped = currentValue.replacingOccurrences(of: "\u{fffc}", with: "")
 
             if currentValueStripped == value {
-                matchesCounter += 1
+                matches += 1
             }
         }
```

Comment:
Collection manipulations can be used here, too.

```swift
let matches = textViews
    .compactMap { ($0.value as? String)?.replacingOccurrences(of: "\u{fffc}", with: "") }
    .filter { $0 == value }
```

Once again, no variables and less code to read.

Reply:
By the way, speaking of reading code. I had to look up what the "\u{fffc}" was.

It might help future readers to encapsulate the value in a constant, e.g.:

```swift
let unicodeReplacementCharacter = "\u{fffc}"
```

Or, one could even define a `String` extension to document what the code above does

```swift
extension String {

    func strippingUnicodeReplacementCharacter() -> String {
        replacingOccurrences(of: "\u{fffc}", with: "")
    }
}
```

Which would make the code here clearer, IMHO:

```swift
let currentValueStripped = currentValue.strippingUnicodeReplacementCharacter()
```

Reply:
@mokagio Again, this is a great improvement. 🙇 I've used `filter` and `compactMap` everywhere whereit wass possible, instead of loops. In the next PR I will try to remove even more loops...